File: ShoppingListLibrary/src/main/java/org/openintents/shopping/SyncSupport.java
package org.openintents.shopping;

import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;

public interface SyncSupport {
    boolean isAvailable();
    void pushListItem(long listId, Cursor cursor);
    void updateListItem(long listId, Uri itemUri, ContentValues values);
    void pushList(Cursor cursor);

    boolean isSyncEnabled();
    void setSyncEnabled(boolean enableSync);
}

File: ShoppingListLibrary/src/main/java/org/openintents/shopping/SyncSupport.kt
package org.openintents.shopping

import android.content.ContentValues
import android.database.Cursor
import android.net.Uri

interface SyncSupport {
    fun isAvailable(): Boolean
    fun pushListItem(listId: Long, cursor: Cursor?)
    fun updateListItem(listId: Long, itemUri: Uri?, values: ContentValues?)
    fun pushList(cursor: Cursor?)
    fun isSyncEnabled(): Boolean
    fun setSyncEnabled(enableSync: Boolean)
}
